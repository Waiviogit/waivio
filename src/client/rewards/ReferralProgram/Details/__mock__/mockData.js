// eslint-disable-next-line import/prefer-default-export
export const mockData = {
  username: 'vallon',
  firstPercent: '50%',
  secondPercent: '40%',
  campaignPercent: '50%',
  indexPercent: '30%',
  referralsPercent: '20%',
  referralDuration: 90,
  additionData: {
    offersReward: 5,
    offersPercent: 5,
    feesValue: '$0.25',
  },
  suspendedTimer: 30,
};

export const mockDataDetails = {
  isAuthenticated: true,
  getReferralDetails: jest.fn(),
  campaignServerPercent: 50,
  indexAbsolutePercent: 60,
  indexServerPercent: 30,
  referralDuration: 90,
  referralServerPercent: 20,
  suspendedTimer: 30,
  username: 'vallon',
};
