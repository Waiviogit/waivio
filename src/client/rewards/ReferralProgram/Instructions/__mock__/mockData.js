// eslint-disable-next-line import/prefer-default-export
export const mockData = {
  authUserName: 'vallon',
  getUserInBlackList: () => {},
  isBlackListUser: false,
  isAuthenticated: true,
  isGuest: false,
  confirmRules: () => {},
  rejectRules: () => {},
  userReferralInfo: () => {},
  referralStatus: 'rejected',
  isStartChangeRules: false,
  isStartGetReferralInfo: false,
};

export const mockDataView = {
  isBlackListUser: false,
  isAuthenticated: true,
  rejectRules: jest.fn(),
  isStartChangeRules: false,
  isStartGetReferralInfo: false,
  handleAgreeRulesCheckbox: jest.fn(),
  handleOkButton: jest.fn(),
  handleCancelButton: jest.fn(),
  currentCopyText: '',
  authUserName: 'vallon',
  isModal: false,
  isGuest: false,
  currentStatus: false,
  setIsCopyButton: jest.fn(),
};

export const mockDataConfirm = {
  isBlackListUser: false,
  isAuthenticated: true,
  rejectRules: jest.fn(),
  isStartChangeRules: false,
  isStartGetReferralInfo: false,
  handleAgreeRulesCheckbox: jest.fn(),
  handleOkButton: jest.fn(),
  handleCancelButton: jest.fn(),
  currentCopyText: '',
  authUserName: 'vallon',
  isModal: false,
  isGuest: false,
  currentStatus: true,
  setIsCopyButton: jest.fn(),
};

export const mockDataIsModal = {
  isBlackListUser: false,
  isAuthenticated: true,
  rejectRules: jest.fn(),
  isStartChangeRules: false,
  isStartGetReferralInfo: false,
  handleAgreeRulesCheckbox: jest.fn(),
  handleOkButton: jest.fn(),
  handleCancelButton: jest.fn(),
  currentCopyText: '',
  authUserName: 'vallon',
  isModal: true,
  isGuest: false,
  currentStatus: true,
  setIsCopyButton: jest.fn(),
};

export const mockDataIsBlackListUser = {
  isBlackListUser: true,
  isAuthenticated: true,
  rejectRules: jest.fn(),
  isStartChangeRules: false,
  isStartGetReferralInfo: false,
  handleAgreeRulesCheckbox: jest.fn(),
  handleOkButton: jest.fn(),
  handleCancelButton: jest.fn(),
  currentCopyText: '',
  authUserName: 'vallon',
  isModal: false,
  isGuest: false,
  currentStatus: false,
  setIsCopyButton: jest.fn(),
};

export const mockDataIsLoadingState = {
  isBlackListUser: false,
  isAuthenticated: true,
  rejectRules: jest.fn(),
  isStartChangeRules: true,
  isStartGetReferralInfo: false,
  handleAgreeRulesCheckbox: jest.fn(),
  handleOkButton: jest.fn(),
  handleCancelButton: jest.fn(),
  currentCopyText: '',
  authUserName: 'vallon',
  isModal: false,
  isGuest: false,
  currentStatus: false,
  setIsCopyButton: jest.fn(),
};
