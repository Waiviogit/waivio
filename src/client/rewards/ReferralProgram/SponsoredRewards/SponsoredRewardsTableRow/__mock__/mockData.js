export const data = {
  username: 'waivio_dsad-asas',
  sponsorName: 'asd09',
  prymaryObjectName: 'myTest',
  prymaryObjectLink: 'cqd-mytest',
  reviewObjectName: 'MyNewTest',
  reviewObjectLink: 'rxt-mynewtest',
  reservationPermlink: 'reserve-ehq58v56iyd',
};

export const sponsorInfo = {
  author_permlink: 'cqd-mytest',
  default_name: 'MyTest',
  object_type: 'restaurant',
  name: 'myTest',
  defaultShowLink: '/object/cqd-mytest',
};

export const dataReview = '13-Oct-2020';
export const userWeight = ' (100%)';

export const sponsor = {
  _id: '5f854d0f5110180f72a5e386',
  payed: false,
  withdraw: null,
  is_demo_account: true,
  recounted: false,
  memo: '',
  userName: 'waivio_dsad-asas',
  sponsor: 'asd09',
  type: 'referral_server_fee',
  app: 'waiviodev/1.0.0',
  details: {
    beneficiaries: [],
    review_permlink: '5rhsvw-review-mytest-mynewtest',
    review_object: {
      author_permlink: 'rxt-mynewtest',
      default_name: 'MyNewTest',
      object_type: 'restaurant',
      name: 'MyNewTest',
      defaultShowLink: '/object/rxt-mynewtest',
    },
    reservation_permlink: 'reserve-ehq58v56iyd',
    main_object: {
      author_permlink: 'cqd-mytest',
      default_name: 'MyTest',
      object_type: 'restaurant',
      name: 'myTest',
      defaultShowLink: '/object/cqd-mytest',
    },
    hiveCurrency: 0.154579,
    commissionWeight: 307.6923,
    payableInDollars: 0.00030915799999999997,
  },
  amount: 0.002,
  createdAt: '2020-10-13T06:45:35.517Z',
  updatedAt: '2020-10-13T06:45:35.517Z',
  __v: 0,
  currentUser: 'vallon',
  balance: 0.002,
};

export const isModalReportOpen = false;
export const closeModalReport = jest.fn();
export const currentAmount = '0.002';
export const currentBalance = '0.002';
