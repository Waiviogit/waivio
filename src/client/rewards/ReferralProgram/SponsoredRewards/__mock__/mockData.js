// eslint-disable-next-line import/prefer-default-export
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import React from 'react';

export const sponsor = [
  {
    _id: '5f854d0f5110180f72a5e386',
  },
];

export const rewardsTitle = (
  <FormattedMessage
    id="sponsored_rewards_title"
    defaultMessage="Sponsored rewards: {username}"
    values={{
      username: (
        <Link to={`/@vallon`}>
          <span className="title-link">vallon</span>
        </Link>
      ),
    }}
  />
);

export const mockIntl = {
  formatMessage: jest.fn(),
};
