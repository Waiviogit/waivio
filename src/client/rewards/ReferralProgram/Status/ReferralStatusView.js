import React from 'react';
import PropTypes from 'prop-types';
import { referralStatusContent } from '../ReferralTextHelper';
import { handleLoadMoreUserStatusCards, handleStatusDaysLeft } from '../ReferralHelper';
import ReferralStatusSort from './ReferralStatusSort/ReferralStatusSort';
import ReduxInfiniteScroll from '../../../vendor/ReduxInfiniteScroll';
import Loading from '../../../components/Icon/Loading';
import ReferralUserCard from './UserStatusCard/ReferralUserCard';

const ReferralStatusView = ({ propsData }) => {
  const {
    username,
    isAuthUser,
    currentUserCards,
    hasMore,
    isErrorLoading,
    isLoadingMoreUserCards,
    getMoreUserCards,
    sortBy,
    setSortBy,
    history,
  } = propsData;
  const data = { username, currentUserCards };
  const { statusTitle, statusDescription, statusCount, statusPaymentText } = referralStatusContent(
    data,
  );

  const handleLoadMore = () => {
    const values = {
      username,
      currentUserCards,
      isLoadingMoreUserCards,
      getMoreUserCards,
    };
    return handleLoadMoreUserStatusCards(values);
  };

  return (
    <React.Fragment>
      {isAuthUser && (
        <div className="ReferralStatus">
          <h2 className="ReferralStatus__title">{statusTitle}</h2>
          <div className="ReferralStatus__description">{statusDescription}</div>
          <div className="ReferralStatus__container">
            <span className="ReferralStatus__total-count">{statusCount}</span>
            <span className="ReferralStatus__sort-by">
              <ReferralStatusSort handleSortChange={setSortBy} sortBy={sortBy} />
            </span>
          </div>
          <div className="ReferralStatus__user-cards">
            {currentUserCards && (
              <ReduxInfiniteScroll
                loadMore={handleLoadMore}
                hasMore={hasMore}
                elementIsScrollable={false}
                threshold={500}
                loader={
                  !isErrorLoading && (
                    <div className="ReferralStatus__user-cards__loader">
                      <Loading />
                    </div>
                  )
                }
              >
                {currentUserCards.map(userCard => (
                  <ReferralUserCard
                    key={`${userCard.name}${userCard.started}`}
                    alias={userCard.alias}
                    username={userCard.name}
                    daysLeft={handleStatusDaysLeft(userCard.daysLeft)}
                    history={history}
                  />
                ))}
              </ReduxInfiniteScroll>
            )}
          </div>
          <div className="ReferralStatus__payment-text">{statusPaymentText}</div>
        </div>
      )}
    </React.Fragment>
  );
};

ReferralStatusView.propTypes = {
  propsData: PropTypes.shape({
    username: PropTypes.string,
    isAuthUser: PropTypes.bool,
    currentUserCards: PropTypes.shape(),
    hasMore: PropTypes.bool,
    isErrorLoading: PropTypes.bool,
    isLoadingMoreUserCards: PropTypes.bool,
    getMoreUserCards: PropTypes.func,
    sortBy: PropTypes.string,
    setSortBy: PropTypes.func,
    history: PropTypes.shape(),
  }),
};

ReferralStatusView.defaultProps = {
  propsData: PropTypes.shape({
    username: '',
    isAuthUser: false,
    currentUserCards: [],
    hasMore: false,
    isErrorLoading: false,
    isLoadingMoreUserCards: false,
    getMoreUserCards: () => {},
    sortBy: '',
    setSortBy: () => {},
    history: {},
  }),
};

export default ReferralStatusView;
