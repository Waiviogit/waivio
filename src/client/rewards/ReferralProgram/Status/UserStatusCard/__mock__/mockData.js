// eslint-disable-next-line import/prefer-default-export
export const mockData = {
  username: 'vallon',
  alias: '',
  daysLeft: 277,
  history: {
    push: jest.fn(),
  },
};
