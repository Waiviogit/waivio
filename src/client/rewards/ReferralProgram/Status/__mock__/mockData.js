// eslint-disable-next-line import/prefer-default-export
export const mockData = {
  username: 'vallon',
  isAuthUser: true,
  currentUserCards: [
    {
      alias: 'Waivio Service',
      guideName: 'sor31',
      lastCreatedAt: '2020-08-31T07:40:03.623Z',
      payable: 0.008,
      payed: false,
    },
    {
      alias: 'Waivio Service',
      guideName: 'asd09',
      lastCreatedAt: '2020-10-13T15:07:13.471Z',
      payable: 0.006,
      payed: false,
    },
    {
      alias: '',
      guideName: 'vallon',
      lastCreatedAt: '2020-06-22T07:39:21.654Z',
      payable: 0.004,
      payed: false,
    },
  ],
  hasMore: false,
  isErrorLoading: false,
  isLoadingMoreUserCards: false,
  getMoreUserCards: jest.fn(),
  sortBy: 'recency',
  setSortBy: jest.fn(),
  history: {},
};
