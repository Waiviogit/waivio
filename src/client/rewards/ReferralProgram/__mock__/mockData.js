// eslint-disable-next-line import/prefer-default-export
// import { GET_USER_REFERRAL_INFO } from '../ReferralActions';

export const mockDataInitialState = {
  isStartLoadingReferralDetails: false,
  campaignServerPercent: null,
  indexAbsolutePercent: null,
  indexServerPercent: null,
  referralDuration: null,
  referralServerPercent: null,
  suspendedTimer: null,
  isUserInWaivioBlackList: false,
  isStartGetReferralInfo: false,
  isStartChangeRules: false,
  referralStatus: '',
  referral: [],
  isChangedRuleSelection: false,
  isGetUsersCards: false,
  hasMoreCards: false,
  userCards: [],
  isLoadingMoreUserCards: false,
  isErrorLoadingMore: false,
  statusSponsoredHistory: [],
};

export const MOCK_GET_USER_REFERRAL_INFO_START = {
  isStartGetReferralInfo: true,
};

export const MOCK_GET_USER_REFERRAL_INFO_ERROR = {
  isStartGetReferralInfo: false,
};

export const MOCK_REFERRAL_GET_ADDITION_FIELDS_START = {
  isChangedRuleSelection: true,
};

export const MOCK_GET_USER_REFERRAL_INFO_SUCCESS = {
  referralStatus: 'activate',
  referral: [],
  isStartGetReferralInfo: false,
};

export const MOCK_GET_USER_REFERRAL_DETAILS_START = {
  isStartLoadingReferralDetails: true,
};

export const MOCK_GET_USER_REFERRAL_DETAILS_SUCCESS = {
  isStartLoadingReferralDetails: false,
  campaignServerPercent: 50,
  indexAbsolutePercent: 60,
  indexServerPercent: 30,
  referralDuration: 90,
  referralServerPercent: 20,
  suspendedTimer: 30,
};

export const MOCK_REFERRAL_GET_ADDITION_FIELDS_SUCCESS = {
  referralStatus: 'activate',
  referral: [],
  isChangedRuleSelection: false,
};

export const MOCK_GET_USER_STATUS_CARDS_START = {
  isGetUsersCards: true,
};

export const MOCK_GET_USER_STATUS_CARDS_SUCCESS = {
  hasMoreCards: false,
  campaignServerPercent: null,
  isGetUsersCards: false,
  isErrorLoadingMore: false,
  isLoadingMoreUserCards: false,
  userCards: [
    {
      alias: '',
      name: 'waivio_dsad-asas',
      started: '2020-10-10T13:20:12.267Z',
      ended: '2021-01-08T13:20:12.267Z',
      daysLeft: -284,
    },
    {
      alias: '',
      name: 'waivio_vallon-kings',
      started: '2020-10-09T12:55:22.618Z',
      ended: '2021-01-07T12:55:22.619Z',
      daysLeft: -285,
    },
    {
      alias: '',
      name: 'waivio_sergei-sergei',
      started: '2020-10-09T12:50:06.394Z',
      ended: '2021-01-07T12:50:06.394Z',
      daysLeft: -285,
    },
  ],
};

export const MOCK_GET_MORE_USER_STATUS_CARDS_START = {
  isLoadingMoreUserCards: true,
  isErrorLoadingMore: false,
};

export const MOCK_GET_MORE_USER_STATUS_CARDS_ERROR = {
  isErrorLoadingMore: true,
};

export const MOCK_REFERRAL_CONFIRM_RULES_START = {
  isStartChangeRules: true,
};

export const MOCK_REFERRAL_CONFIRM_RULES_SUCCESS = {
  isStartChangeRules: false,
};

export const MOCK_REFERRAL_REJECT_RULES_START = {
  isStartChangeRules: true,
};

export const MOCK_REFERRAL_REJECT_RULES_SUCCESS = {
  isStartChangeRules: false,
};

export const sponsor = {
  details: {
    beneficiaries: [{ account: 'waivio', weight: 300 }],
    commissionWeight: null,
    hiveCurrency: 0.238689,
    main_object: {
      author_permlink: 'vvr-floret',
      defaultShowLink: '/object/vvr-floret',
      default_name: 'Floret',
      name: 'Floret',
      object_type: 'restaurant',
    },
    payableInDollars: 0.0009308871,
    reservation_permlink: 'reserve-hypzi77o2nh',
    review_object: {
      author_permlink: 'tbf-clear-tea',
      defaultShowLink: '/object/tbf-clear-tea',
      default_name: 'Clear tea',
      name: 'Clear tea',
      object_type: 'drink',
    },
    review_permlink: '4cmiet-review-floret-clear-tea',
  },
};

export const sponsorWithoutBeneficiaries = {
  details: {},
};

export const handleLoadMoreData = {
  username: 'vallon',
  currentUserCards: [
    {
      alias: 'Waivio Service',
      guideName: 'sor31',
      lastCreatedAt: '2020-08-31T07:40:03.623Z',
      payable: 0.008,
      payed: false,
    },
    {
      alias: 'Waivio Service',
      guideName: 'asd09',
      lastCreatedAt: '2020-10-13T15:07:13.471Z',
      payable: 0.006,
      payed: false,
    },
    {
      alias: '',
      guideName: 'vallon',
      lastCreatedAt: '2020-06-22T07:39:21.654Z',
      payable: 0.004,
      payed: false,
    },
    {
      alias: 'Waivio Service',
      guideName: 'sor32',
      lastCreatedAt: '2020-08-31T07:40:03.623Z',
      payable: 0.008,
      payed: false,
    },
    {
      alias: 'Waivio Service',
      guideName: 'asd093',
      lastCreatedAt: '2020-10-13T15:07:13.471Z',
      payable: 0.006,
      payed: false,
    },
    {
      alias: '',
      guideName: 'vallon',
      lastCreatedAt: '2020-06-22T07:39:21.654Z',
      payable: 0.004,
      payed: false,
    },
    {
      alias: 'Waivio Service',
      guideName: 'sor31',
      lastCreatedAt: '2020-08-31T07:40:03.623Z',
      payable: 0.008,
      payed: false,
    },
    {
      alias: 'Waivio Service',
      guideName: 'asd09',
      lastCreatedAt: '2020-10-13T15:07:13.471Z',
      payable: 0.006,
      payed: false,
    },
    {
      alias: '',
      guideName: 'vallon',
      lastCreatedAt: '2020-06-22T07:39:21.654Z',
      payable: 0.004,
      payed: false,
    },
    {
      alias: 'Waivio Service',
      guideName: 'sor32as',
      lastCreatedAt: '2020-08-31T07:40:03.623Z',
      payable: 0.008,
      payed: false,
    },
    {
      alias: 'Waivio Service',
      guideName: 'asd09s3',
      lastCreatedAt: '2020-10-13T15:07:13.471Z',
      payable: 0.006,
      payed: false,
    },
    {
      alias: '',
      guideName: 'vallon',
      lastCreatedAt: '2020-06-22T07:39:21.654Z',
      payable: 0.004,
      payed: false,
    },
  ],
  isLoadingMoreUserCards: false,
  getMoreUserCards: jest.fn(),
  sort: 'recency',
};
