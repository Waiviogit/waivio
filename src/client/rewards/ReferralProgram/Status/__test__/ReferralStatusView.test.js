import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import { shallow } from 'enzyme';
import { mockData } from '../__mock__/mockData';
import ReferralStatusView from '../ReferralStatusView';

describe('ReferralStatusView', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = mockData;
    wrapper = shallow(
      <BrowserRouter>
        <IntlProvider locale="en">
          <ReferralStatusView propsData={props} />
        </IntlProvider>
      </BrowserRouter>,
    );
  });

  afterEach(() => jest.clearAllMocks());

  it("should create ReferralStatusView's snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
