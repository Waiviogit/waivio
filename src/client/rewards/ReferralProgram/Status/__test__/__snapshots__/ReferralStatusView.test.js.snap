// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferralStatusView should create ReferralStatusView's snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserRouter>
    <IntlProvider
      locale="en"
    >
      <ReferralStatusView
        propsData={
          Object {
            "currentUserCards": Array [
              Object {
                "alias": "Waivio Service",
                "guideName": "sor31",
                "lastCreatedAt": "2020-08-31T07:40:03.623Z",
                "payable": 0.008,
                "payed": false,
              },
              Object {
                "alias": "Waivio Service",
                "guideName": "asd09",
                "lastCreatedAt": "2020-10-13T15:07:13.471Z",
                "payable": 0.006,
                "payed": false,
              },
              Object {
                "alias": "",
                "guideName": "vallon",
                "lastCreatedAt": "2020-06-22T07:39:21.654Z",
                "payable": 0.004,
                "payed": false,
              },
            ],
            "getMoreUserCards": [MockFunction],
            "hasMore": false,
            "history": Object {},
            "isAuthUser": true,
            "isErrorLoading": false,
            "isLoadingMoreUserCards": false,
            "setSortBy": [MockFunction],
            "sortBy": "recency",
            "username": "vallon",
          }
        }
      />
    </IntlProvider>
  </BrowserRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <IntlProvider
        locale="en"
      >
        <ReferralStatusView
          propsData={
            Object {
              "currentUserCards": Array [
                Object {
                  "alias": "Waivio Service",
                  "guideName": "sor31",
                  "lastCreatedAt": "2020-08-31T07:40:03.623Z",
                  "payable": 0.008,
                  "payed": false,
                },
                Object {
                  "alias": "Waivio Service",
                  "guideName": "asd09",
                  "lastCreatedAt": "2020-10-13T15:07:13.471Z",
                  "payable": 0.006,
                  "payed": false,
                },
                Object {
                  "alias": "",
                  "guideName": "vallon",
                  "lastCreatedAt": "2020-06-22T07:39:21.654Z",
                  "payable": 0.004,
                  "payed": false,
                },
              ],
              "getMoreUserCards": [MockFunction],
              "hasMore": false,
              "history": Object {},
              "isAuthUser": true,
              "isErrorLoading": false,
              "isLoadingMoreUserCards": false,
              "setSortBy": [MockFunction],
              "sortBy": "recency",
              "username": "vallon",
            }
          }
        />
      </IntlProvider>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ReferralStatusView
          propsData={
            Object {
              "currentUserCards": Array [
                Object {
                  "alias": "Waivio Service",
                  "guideName": "sor31",
                  "lastCreatedAt": "2020-08-31T07:40:03.623Z",
                  "payable": 0.008,
                  "payed": false,
                },
                Object {
                  "alias": "Waivio Service",
                  "guideName": "asd09",
                  "lastCreatedAt": "2020-10-13T15:07:13.471Z",
                  "payable": 0.006,
                  "payed": false,
                },
                Object {
                  "alias": "",
                  "guideName": "vallon",
                  "lastCreatedAt": "2020-06-22T07:39:21.654Z",
                  "payable": 0.004,
                  "payed": false,
                },
              ],
              "getMoreUserCards": [MockFunction],
              "hasMore": false,
              "history": Object {},
              "isAuthUser": true,
              "isErrorLoading": false,
              "isLoadingMoreUserCards": false,
              "setSortBy": [MockFunction],
              "sortBy": "recency",
              "username": "vallon",
            }
          }
        />,
        "locale": "en",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "propsData": Object {
            "currentUserCards": Array [
              Object {
                "alias": "Waivio Service",
                "guideName": "sor31",
                "lastCreatedAt": "2020-08-31T07:40:03.623Z",
                "payable": 0.008,
                "payed": false,
              },
              Object {
                "alias": "Waivio Service",
                "guideName": "asd09",
                "lastCreatedAt": "2020-10-13T15:07:13.471Z",
                "payable": 0.006,
                "payed": false,
              },
              Object {
                "alias": "",
                "guideName": "vallon",
                "lastCreatedAt": "2020-06-22T07:39:21.654Z",
                "payable": 0.004,
                "payed": false,
              },
            ],
            "getMoreUserCards": [MockFunction],
            "hasMore": false,
            "history": Object {},
            "isAuthUser": true,
            "isErrorLoading": false,
            "isLoadingMoreUserCards": false,
            "setSortBy": [MockFunction],
            "sortBy": "recency",
            "username": "vallon",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <IntlProvider
          locale="en"
        >
          <ReferralStatusView
            propsData={
              Object {
                "currentUserCards": Array [
                  Object {
                    "alias": "Waivio Service",
                    "guideName": "sor31",
                    "lastCreatedAt": "2020-08-31T07:40:03.623Z",
                    "payable": 0.008,
                    "payed": false,
                  },
                  Object {
                    "alias": "Waivio Service",
                    "guideName": "asd09",
                    "lastCreatedAt": "2020-10-13T15:07:13.471Z",
                    "payable": 0.006,
                    "payed": false,
                  },
                  Object {
                    "alias": "",
                    "guideName": "vallon",
                    "lastCreatedAt": "2020-06-22T07:39:21.654Z",
                    "payable": 0.004,
                    "payed": false,
                  },
                ],
                "getMoreUserCards": [MockFunction],
                "hasMore": false,
                "history": Object {},
                "isAuthUser": true,
                "isErrorLoading": false,
                "isLoadingMoreUserCards": false,
                "setSortBy": [MockFunction],
                "sortBy": "recency",
                "username": "vallon",
              }
            }
          />
        </IntlProvider>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ReferralStatusView
            propsData={
              Object {
                "currentUserCards": Array [
                  Object {
                    "alias": "Waivio Service",
                    "guideName": "sor31",
                    "lastCreatedAt": "2020-08-31T07:40:03.623Z",
                    "payable": 0.008,
                    "payed": false,
                  },
                  Object {
                    "alias": "Waivio Service",
                    "guideName": "asd09",
                    "lastCreatedAt": "2020-10-13T15:07:13.471Z",
                    "payable": 0.006,
                    "payed": false,
                  },
                  Object {
                    "alias": "",
                    "guideName": "vallon",
                    "lastCreatedAt": "2020-06-22T07:39:21.654Z",
                    "payable": 0.004,
                    "payed": false,
                  },
                ],
                "getMoreUserCards": [MockFunction],
                "hasMore": false,
                "history": Object {},
                "isAuthUser": true,
                "isErrorLoading": false,
                "isLoadingMoreUserCards": false,
                "setSortBy": [MockFunction],
                "sortBy": "recency",
                "username": "vallon",
              }
            }
          />,
          "locale": "en",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "propsData": Object {
              "currentUserCards": Array [
                Object {
                  "alias": "Waivio Service",
                  "guideName": "sor31",
                  "lastCreatedAt": "2020-08-31T07:40:03.623Z",
                  "payable": 0.008,
                  "payed": false,
                },
                Object {
                  "alias": "Waivio Service",
                  "guideName": "asd09",
                  "lastCreatedAt": "2020-10-13T15:07:13.471Z",
                  "payable": 0.006,
                  "payed": false,
                },
                Object {
                  "alias": "",
                  "guideName": "vallon",
                  "lastCreatedAt": "2020-06-22T07:39:21.654Z",
                  "payable": 0.004,
                  "payed": false,
                },
              ],
              "getMoreUserCards": [MockFunction],
              "hasMore": false,
              "history": Object {},
              "isAuthUser": true,
              "isErrorLoading": false,
              "isLoadingMoreUserCards": false,
              "setSortBy": [MockFunction],
              "sortBy": "recency",
              "username": "vallon",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
